### DISCUSSION TITLE
How to restart i3 when it no longer responds to keybindings?

### DISCUSSION THREAD

**Discussion by medwatt** [i3/i3#4867] [Q&A]
I use i3 as the windows manager in KDE. It works quite well expect in those situations where something happens that causes the windows to no longer be tiled, the i3 bar gone, and i3 no longer responding to keybindings. My usually remedy is to switch to start a new tty and logout from my current i3 session. However, I lose all unsaved changes using this method. I wonder if there's anther way I can restart the windows manager without logging out. Is there a way?

**Comment by orestisfl** [comment]
Switch to a different tty and use `DISPLAY=:0 i3 restart`

But maybe you can provide more info to figure out why a restart is needed in the first place

**Reply by medwatt** [reply]
Thanks for the reply. As I said, this only happens when using i3 inside of KDE. It is something that I cannot replicate consistently. It tends to happen a lot when I'm using zotero (document management tool). I haven't experienced this issue yet when using standalone i3. I would like to continue using i3 on top of KDE because of a lot of nice things like getting notification via kde connect, display management, proper theming, etc.

I has tried using standalone i3 and I was looking for a way to easily connect my second monitor. For some reason, after enabling my second monitor, i3 stops recognizing its own keybindings and instead recognizes does of KDE. I don't know why this happens since I am using standalone i3.

Here's the snippet of the code I am using:

```bash
set $display Display Mode: clone (c), right (r)
mode "$display" {
    bindsym c exec xrandr --output $secondMonitor --same-as $firstMonitor --auto 2> /dev/null, mode "default"
    bindsym r exec xrandr --output $secondMonitor --right-of $firstMonitor --auto 2> /dev/null, mode "default"

    # back to normal: Enter or Escape
    bindsym Return mode "default"
    bindsym Escape mode "default"
}
bindsym $super+XF86Display mode "$display"
```
